src/components/editor/editor.tsx

```diff
@@ -1,214 +1,88 @@
-  //         if (def.name) {
-  //           operationName = def.name.value;
-  //         }
-  //         break;
-
-  //       default:
-  //         operationKind = 'unknown';
-  //         operationName = 'unknown';
-  //         break;
-  //     }
-
-  //     const selector = `.graphiql-explorer-root #${operationKind}-${operationName}`;
-
-  //     const el = document.querySelector(selector);
-  //     el && el.scrollIntoView();
-  //   },
-  //   [query]
-  // );
-  // Extra configuration for editor
-  // useEffect(() => {
-  //   const editor = graphiqlRef.current?.getQueryEditor();
-  //   // const existingOptions = editor?.getOption('extraKeys');
-  //   // editor?.setOption('extraKeys', {
-  //   //   // ...(typeof existingOptions !== 'string' ? existingOptions : {}),
-  //   //   'Shift-Alt-LeftClick': handleInspectOperation,
-  //   // });
-  // }, [graphiqlRef, handleInspectOperation]);
-
-  const handleToggleExplorer = useCallback(() => {
-    setExplorerIsOpen((prevState) => !prevState);
-  }, []);
+    setQuery(props.initialQuery);
+  }, [props.initialQuery]);
 
   return (
     <div>
-      <div className="graphiql-container">
-        <GraphiQLExplorer
-          schema={schema}
-          query={query ?? ''}
-          onEdit={setQuery}
-          onRunOperation={(operationName) =>
-            graphiqlRef.current?.handleRunQuery(operationName)
-          }
-          explorerIsOpen={explorerIsOpen}
-          onToggleExplorer={handleToggleExplorer}
-        />
+      <QueryContext.Provider value={context}>
         <GraphiQL
-          ref={graphiqlRef}
           fetcher={fetcher}
-          schema={schema}
           query={query}
-          onEditQuery={setQuery}
-          storage={{
-            getItem: (key) =>
-              window.localStorage.getItem(`${key}:${props.target}`),
-            setItem: (key, value) =>
-              window.localStorage.setItem(`${key}:${props.target}`, value),
-            removeItem: (key) =>
-              window.localStorage.removeItem(`${key}:${props.target}`),
-            length: 1, // no idea what this is useful for.
-          }}
-        >
-          <GraphiQL.Toolbar>
-            <GraphiQL.Button
-              onClick={() => {
-                graphiqlRef.current?.handlePrettifyQuery();
-              }}
-              label="Prettify"
-              title="Prettify Query (Shift-Ctrl-P)"
-            />
-            <GraphiQL.Button
-              onClick={() => {
-                graphiqlRef.current?.handleToggleHistory();
-              }}
-              label="History"
-              title="Show History"
-            />
-            <GraphiQL.Button
-              onClick={handleToggleExplorer}
-              label="Explorer"
-              title="Toggle Explorer"
-            />
-          </GraphiQL.Toolbar>
-        </GraphiQL>
-      </div>
+          onEditQuery={(newQuery) => setQuery(newQuery)}
+          plugins={[explorerPlugin]}
```

Comment:
The plugin explorer has been "copied" from here: https://github.com/graphql/graphiql/tree/main/examples/graphiql-explorer